synapse/storage/schema/main/delta/72/03thread_notifications.sql

```diff
@@ -0,0 +1,27 @@
+/* Copyright 2022 The Matrix.org Foundation C.I.C
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ALTER TABLE event_push_actions_staging
+  ADD COLUMN thread_id TEXT NOT NULL DEFAULT '';
+
+ALTER TABLE event_push_actions
+  ADD COLUMN thread_id TEXT NOT NULL DEFAULT '';
+
+ALTER TABLE event_push_summary
+  ADD COLUMN thread_id TEXT NOT NULL DEFAULT '';
```

Comment:
Does this lock the table on PostgreSQL because we're setting a default?

Might need to use `NULL`, but that doesn't work with `GROUP BY` (since `NULL` is considered != to each other).

Reply:
From reading a bit online this might be incorrect and `GROUP BY` might work OK with nulls, to investigate!

Reply:
Looking at this again, the reasoning for wanting this to be non-null is because we add a unique index to `event_push_summary` on room ID / user ID / thread ID.

I think I just need another constraint for that though.

Reply:
2e85ec6e3077e002cdadaa0199195a69adc35287 implements this.